src/ape/contracts/base.py

```diff
@@ -113,9 +122,62 @@ def __repr__(self) -> str:
         abis = sorted(self.abis, key=lambda abi: len(abi.inputs or []))
         return abis[-1].signature
 
+    def encode_input(self, *args) -> HexBytes:
+        args = self._convert_tuple(args)
+        selected_abi = _select_method_abi(self.abis, args)
+        ecosystem = self.provider.network.ecosystem
+        encoded_calldata = ecosystem.encode_calldata(selected_abi, *args)
+        method_id = ecosystem.get_method_selector(selected_abi)
+        return HexBytes(method_id + encoded_calldata)
+
+    def decode_input(self, calldata: bytes) -> Tuple[str, Dict[str, Any]]:
```

Comment:
Does this do similar logic to what Brownie does?

Reply:
Yes but the second part of the return tuple's format is a little different.
Brownie returns a tuple yes, but the second part is a list of decoded values.
Here, it is a dict of input names (or stringified index if not found) to decoded values.